pkg/translator/prometheusremotewrite/number_data_points_v2_test.go

```diff
@@ -0,0 +1,162 @@
+			name: "int_gauge",
+			metric: func() pmetric.Metric {
+				return getIntGaugeMetric(
+					"test",
+					pcommon.NewMap(),
+					1, ts,
+				)
+			},
+			want: func() map[uint64]*writev2.TimeSeries {
+				labels := labels.Labels{
+					labels.Label{
+						Name:  labels.MetricName,
+						Value: "test",
+					},
+				}
+				return map[uint64]*writev2.TimeSeries{
+					labels.Hash(): {
+						LabelsRefs: []uint32{1, 2},
+						Samples: []writev2.Sample{
+							{Timestamp: convertTimeStamp(pcommon.Timestamp(ts)), Value: 1},
+						},
+					},
+				}
+			},
+		},
+		{
+			name: "double_gauge",
+			metric: func() pmetric.Metric {
+				return getDoubleGaugeMetric(
+					"test",
+					pcommon.NewMap(),
+					1.5, ts,
+				)
+			},
+			want: func() map[uint64]*writev2.TimeSeries {
+				labels := labels.Labels{
+					labels.Label{
+						Name:  labels.MetricName,
+						Value: "test",
+					},
+				}
+				return map[uint64]*writev2.TimeSeries{
+					labels.Hash(): {
+						LabelsRefs: []uint32{1, 2},
+						Samples: []writev2.Sample{
+							{Timestamp: convertTimeStamp(pcommon.Timestamp(ts)), Value: 1.5},
+						},
+					},
+				}
+			},
+		},
+		{
+			name: "gauge with staleNaN",
+			metric: func() pmetric.Metric {
+				return getIntGaugeMetric(
+					"staleNaN",
+					pcommon.NewMap(),
+					1, ts,
+				)
+			},
+			want: func() map[uint64]*writev2.TimeSeries {
+				labels := labels.Labels{
+					labels.Label{
+						Name:  labels.MetricName,
+						Value: "staleNaN",
+					},
+				}
+				return map[uint64]*writev2.TimeSeries{
+					labels.Hash(): {
+						LabelsRefs: []uint32{1, 2},
+						Samples: []writev2.Sample{
+							{Timestamp: convertTimeStamp(pcommon.Timestamp(ts)), Value: math.Float64frombits(value.StaleNaN)},
+						},
+					},
+				}
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			metric := tt.metric()
+			converter := newPrometheusConverterV2()
+			converter.addGaugeNumberDataPoints(metric.Gauge().DataPoints(), metric.Name())
+			w := tt.want()
+
+			diff := cmp.Diff(w, converter.unique, cmpopts.EquateNaNs())
+			assert.Empty(t, diff)
+
+		})
+	}
+}
+
+// Right now we are not handling duplicates, the second one will just overwrite the first one as this test case shows
+// TODO handle duplicates
+func TestPrometheusConverterV2_addGaugeNumberDataPointsDuplicate(t *testing.T) {
+	ts := uint64(time.Now().UnixNano())
+	metric1 := getIntGaugeMetric(
+		"test",
+		pcommon.NewMap(),
+		1, ts,
+	)
+	metric2 := getIntGaugeMetric(
+		"test",
+		pcommon.NewMap(),
+		2, ts,
+	)
+	want := func() map[uint64]*writev2.TimeSeries {
+		labels := labels.Labels{
+			labels.Label{
+				Name:  labels.MetricName,
+				Value: "test",
+			},
+		}
+		return map[uint64]*writev2.TimeSeries{
+			labels.Hash(): {
+				LabelsRefs: []uint32{1, 2},
+				Samples: []writev2.Sample{
+					{Timestamp: convertTimeStamp(pcommon.Timestamp(ts)), Value: 2},
+				},
+			},
+		}
+	}
+
+	t.Run("gauge duplicate", func(t *testing.T) {
```

Comment:
t.Run usually is used when we're testing multiple cases in a loop.

There is only one case here so there's no need for t.Run :)

Reply:
fixed